examples/point_transformer_classification.py

```diff
@@ -0,0 +1,213 @@
+
+from torch_cluster import knn_graph
+from torch_cluster import fps
+from torch_scatter import scatter_max
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data/ModelNet10')
+pre_transform, transform = T.NormalizeScale(), T.Compose(
+    [T.SamplePoints(1024)])
+train_dataset = ModelNet(path, '10', True, transform, pre_transform)
+test_dataset = ModelNet(path, '10', False, transform, pre_transform)
+train_loader = DataLoader(train_dataset, batch_size=32, shuffle=True)
+test_loader = DataLoader(test_dataset, batch_size=32, shuffle=False)
+
+
+class TransformerBlock(torch.nn.Module):
+    def __init__(self, in_channels, out_channels):
+        super(TransformerBlock, self).__init__()
+        self.lin_in = Lin(in_channels, in_channels)
+        self.lin_out = Lin(out_channels, out_channels)
+
+        self.pos_nn = Seq(
+            Lin(3, 64),
+            ReLU(),
+            Lin(64, out_channels)
+        )
+
+        self.attn_nn = Seq(
+            Lin(out_channels, 64),
+            ReLU(),
+            Lin(64, out_channels)
+        )
+
+        self.transformer = PointTransformerConv(
+            in_channels,
+            out_channels,
+            pos_nn=self.pos_nn,
+            attn_nn=self.attn_nn
+        )
+
+    def forward(self, x, pos, edge_index):
+        # x_skip = x.clone()
+        x = self.lin_in(x).relu()
+        x = self.transformer(x, pos, edge_index)
+        x = self.lin_out(x).relu()
+        # x = x + x_skip # negatively impacts training
+        return x
+
+
+class TransitionDown(torch.nn.Module):
+    '''
+        Samples the input point cloud by a ratio percentage to reduce
+        cardinality and uses an mlp to augment features dimensionnality
+    '''
+
+    def __init__(self, in_channels, out_channels, ratio=0.25, k=16):
+        super(TransitionDown, self).__init__()
+        self.k = k
+        self.ratio = ratio
+        self.mlp = Seq(
+            Lin(in_channels, out_channels),
+            BN(out_channels),
+            ReLU()
+        )
+
+    def forward(self, x, pos, batch):
+        # FPS sampling
+        id_clusters = fps(pos, ratio=self.ratio, batch=batch)
+
+        # compute for each cluster the k nearest points
+        sub_batch = batch[id_clusters] if batch is not None else None
+
+        # beware of self loop
+        id_k_neighbor = knn(pos, pos[id_clusters], k=self.k,
+                            batch_x=batch, batch_y=sub_batch)
+
+        # transformation of features through a simple MLP
+        x = self.mlp(x)
+
+        # Max pool onto each cluster the features from knn in points
+        x_out, _ = scatter_max(x[id_k_neighbor[1]],
+                               id_k_neighbor[0],
+                               dim_size=id_clusters.size(0),
+                               dim=0)
+
+        # keep only the clusters and their max-pooled features
+        new_pos, new_x = pos[id_clusters], x_out
+        return new_x, new_pos, sub_batch
```

Comment:
Let's call the subset `pos` tensor `sub_pos` as well to make naming consistent.
```suggestion
        return out, sub_pos, sub_batch
```